sssd_test_framework/utils/sssd.py

```diff
@@ -281,6 +304,11 @@ def restart(
         if service == "sssd":
             self.svc.stop("sssd-kcm.service")
 
+        if clean and service == "sssd":
+            self.svc.stop(service)
+            self.clear()
+            return self.svc.start(service, raise_on_error=raise_on_error)
```

Comment:
Just clear is sufficient as in other places, there is no need to stop it explicitly.